File: src/main/java/tn/esprit/fundsphere/Entites/AccountManagment/Account.java
package tn.esprit.fundsphere.Entites.AccountManagment;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.fundsphere.Entites.UserManagment.User;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "account")
public class Account {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    int idAccount;
    int numAccount;
    double balance;
    Long rib;
    LocalDate date;
    @Enumerated(EnumType.STRING)
    TypeAccount typeA;

    @ManyToOne
    User user;



}

File: angular/src/app/core/models/account.ts
import { TypeAccount } from "./type-account";
import { Transaction } from "./transaction";
import { UserShow } from "./user-show";
export class Account {
    
    idAccount!: number;
    numAccount!: number;
    rib!: number;
    balance!: number;
    date!: Date;
    typeA!: TypeAccount;
    transactions?: Transaction[]; 
    user?: UserShow; 

    
}


